RecoBTag/ONNXRuntime/interface/tensor_configs.h

```diff
@@ -28,4 +29,37 @@ namespace deepvertex {
 
 }  // namespace deepvertex
 
+namespace parT {
+
+  enum InputFeatures {
+    kBegin = 0,
+    kChargedCandidates = kBegin,
+    kNeutralCandidates = 1,
+    kVertices = 2,
+    kChargedCandidates4Vec = 3,
+    kNeutralCandidates4Vec = 4,
+    kVertices4Vec = 5,
+    kEnd = 6
+  };
+
+  constexpr unsigned n_cpf_accept = 25;
+  constexpr unsigned n_npf_accept = 25;
+  constexpr unsigned n_sv_accept = 5;
+
+  const std::map<InputFeatures, unsigned int> N_InputFeatures{{kChargedCandidates, 16},
```

Comment:
If defined this way, these global constants should be defined as `inline`.

Given that the keys are adjacent integers starting from 0, the map could be replaced with `constexpr std::array<unsigned int, kEnd>` that would avoid dynamic memory allocations, be faster to access, and be `inline` implicitly. (nowadays with C++20 also `constexpr std::vector` should work, but `std::array` seems easy enough here).

Reply:
Hi @makortel somehow I received warning being treated as error when compiling std::array, and it can be fixed by using double braces...  `/depot/cms/private/users/yao317/SONIC/tmp/CMSSW_14_1_0_pre0/src/RecoBTag/ONNXRuntime/interface/tensor_configs.h:49:67: error: missing braces around initializer for 'std::__array_traits<unsigned int, 6>::_Type' {aka 'unsigned int [6]'} [-Werror=missing-braces]
   49 |   inline constexpr std::array<unsigned int, 6> dummy { 1,1,1,1,1,1};` if you are fine with it, I will update my branch with inline and constexpr of all the variables in namespace ParT.

Reply:
@kpedro88 let me know if you have any opinion on this strange warning-error message.

Reply:
@y19y19 you need to put an extra set of curly braces. This is just a C++ syntax requirement. i.e.: `constexpr std::array<unsigned int, 6> dummy {{ 1,1,1,1,1,1}};`